hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -650,6 +647,11 @@ public OutputStream createFile(final Path path,
       if (isAppendBlobKey(path.toString())) {
         isAppendBlob = true;
       }
+      if (!isNamespaceEnabled) {
```

Comment:
This code isn't merged with Anuj's hence initially written so ?
Expectation is that this will become AbfsClientHandler.getClient().method call and no direct checks for is instance of BlobClient or DFSClient per API operation.

Reply:
AbfsClientHandler can have a method to return ingress client.

Reply:
createClient is obtained from getClient method based on ingrss type only and we need to do the marker blobs creation only for blob client and hence this instanceof check. Earlier design was to have an empty method inside the dfsclient class and blobclient would have the actual implementation but based on earlier reviews if a method is unique to a client it should be in that class only so I made the change and hence we need this demarcation

Reply:
Have updated the code in AbfsBlobClient class